TopMovies/View/TopMoviesVC/MovieCategoryTableViewCell.swift

```diff
@@ -7,29 +7,44 @@
 
 import UIKit
 import SnapKit
+// MARK: - MovieCategoryTableViewCellDelegate
+protocol MovieCategoryTableViewCellDelegate: class {
+  func seeAllButtonClicked(_ categoryId: MovieCategory.ID)
+}
 // MARK: - Props struct -
 struct MovieCategoryProps {
+  let categoryId: MovieCategory.ID
   let categoryNameText: String
   let movies: [MovieCollectionProps]
 }
 extension MovieCategoryProps {
-  init?(categoryNameText: String?, movies: [MovieCollectionProps?]) {
+  init?(categoryId: MovieCategory.ID,
+        categoryNameText: String?,
+        movies: [MovieCollectionProps?]) {
     guard let categoryNameText = categoryNameText else { return nil }
+    self.categoryId = categoryId
     self.categoryNameText = categoryNameText
     self.movies = movies.compactMap{ $0 }
   }
 }
 // MARK: - Cell class -
-class MovieCategoryTableViewCell: UITableViewCell {
+final class MovieCategoryTableViewCell: UITableViewCell {
+  weak var delegate: MovieCategoryTableViewCellDelegate?
+  private var categoryId: MovieCategory.ID?
   private let categoryNameLabel = UILabel()
-  private let moviesCollectionView = UICollectionView(frame: CGRect(), collectionViewLayout: UICollectionViewLayout())
-  private let movieCollectionViewCellIdentifier = String(describing: MovieCollectionViewCell.self)
+  private let moviesCollectionView
+    = UICollectionView(frame: CGRect(), collectionViewLayout: UICollectionViewLayout())
+  private let movieCollectionViewCellIdentifier
+    = String(describing: MovieCollectionViewCell.self)
+  private let seeAllMoviesCollectionCellIdentifier
+    = String(describing: SeeAllMoviesCollectionViewCell.self)
   private let movieCollectionInteritemSpacing: CGFloat = 8.0
   private let infoStackView = UIStackView()
   private var moviesProps = [MovieCollectionProps]()
   
   // MARK: Cell configuration
   public func configureWith(props: MovieCategoryProps) {
+    categoryId = props.categoryId
     categoryNameLabel.text = props.categoryNameText
     moviesProps = props.movies
   }
```

Comment:
Since you store `moviesProps`, maybe we just store the entire MovieCategoryProps value? To remove multiple variables like id, movesProps etc.